### PR TITLE
HPO terms input, display warning popups, Variant filtering page and other fixes

### PR DESCRIPTION
**Pull request by seeta-ramaraju**
This PR includes changes for the Jira tasks [G2P-188](https://www.ebi.ac.uk/panda/jira/browse/G2P-188), [G2P-191](https://www.ebi.ac.uk/panda/jira/browse/G2P-191), [G2P-171](https://www.ebi.ac.uk/panda/jira/browse/G2P-171) and other fixes/improvements in add g2p record page and update draft g2p record page.

### CODE REVIEW

**Comment by dglemos** [comment]
Tested the create and update pages. It works as expected.

For the next step I would recommend to update the following in the update page:
- when inserting a new pmid, the page should add the new id to the existing list instead of replacing the list.
- same for the phenotypes list - add hpo terms to the existing list of hpo terms.

**Review by dglemos — APPROVED** [review]

**Review by olaaustine — COMMENTED** [review]
Minor comments about wordings

File: src/views/AddLocusGeneDiseaseView.vue [hunk 1]
```diff
@@ -144,41 +148,102 @@ export default {
...
+      // if hpoTermsInput is empty then set isHpoTermsValid to false and dont continue further
+      if (this.hpoTermsInputHelper[pmid].hpoTermsInput.trim() === "") {
+        this.hpoTermsInputHelper[pmid].isHpoTermsValid = false;
+        return;
+      }
+      // if hpoTermsInput is not empty then continue further
+      this.hpoTermsInputHelper[pmid].isHpoTermsValid = true;
+      this.hpoTermsInputHelper[pmid].hpoTermsErrorMsg = "";
+      this.hpoTermsInputHelper[pmid].isHpoTermsDataLoading = true;
+      let hpoTermsListStr = this.hpoTermsInputHelper[pmid].hpoTermsInput
```
Comment (olaaustine) [inline src/views/AddLocusGeneDiseaseView.vue:211]: There is a bug here, the split should also make sure there is no space between semi colons and the next HP0 term. Thank you

Comment (seeta-ramaraju) [reply src/views/AddLocusGeneDiseaseView.vue:211]: taken care, the new change will trim any space around the hpo term and between semi colons

**Comment by olaaustine** [comment]
Hi Raju, just a minor comment about updating hpo input handling for Update page also. Thank you

File: src/views/VariantFilteringView.vue [hunk 2]
```diff
@@ -0,0 +1,402 @@
...
+        target="_blank"
+        style="text-decoration: none"
+        >Ensembl VEP</a
+      >) predicts the molecular consequence of a variant and reports relevant
+      pathogenicity predictions and information from reference databases. The
+      Ensembl VEP G2P plugin (VEP-G2P) filters variant genotypes from
+      exome/genome wide sequencing using knowledge encoded in the G2P database
+      to identify likely disease-causing genes.
+    </p>
+    <h6 class="pb-2">How the Ensembl VEP G2P plugin works:</h6>
```
Comment (olaaustine) [inline src/views/VariantFilteringView.vue:19]: Suggesting this is bold

**Comment by seeta-ramaraju** [comment]
> Hi @seeta-ramaraju , just a minor comment about updating hpo input handling for Update page also. Thank you

Taken care in latest commit

File: src/views/VariantFilteringView.vue (hunk 2 above)
Comment (seeta-ramaraju) [reply src/views/VariantFilteringView.vue:19]: taken care in latest commit

**Review by olaaustine — APPROVED** [review]
LGTM

**Pull request merged**